File: Bottle.scala

object Bottle {
    def main(args: Array[String]): Unit = {

        (99 to 1 by -1).par foreach { n =>
            println(
                f"$n%d bottles of beer on the wall. Take one down.\n" +
                f"${n - 1}%d bottles of beer on the wall\n")
        }

        Thread sleep 20000
    }
}

File: bottle.js
"use strict";


function sleep(time) {
	return new Promise((resolve) => setTimeout(resolve, time));
}


{
	let numBottles = 99;
	let word = "bottles";

	while (numBottles > 0) {
		if (numBottles == 1) {
			word = "bottle";
		}
		console.log(`${numBottles} ${word} of beer on the wall. Take one down.`);
		numBottles -= 1;
		if (numBottles > 0) {
			console.log(`${numBottles} ${word} of beer on the wall.`);

		} else {
			sleep(20000);
			console.log("No more bottles of beer on the wall"); 
		}
	}
}
